CMakeLists.txt

```diff
@@ -598,7 +598,9 @@ if(HPX_WITH_NETWORKING)
   endif()
 endif(HPX_WITH_NETWORKING)
 
-if((HPX_WITH_NETWORKING AND HPX_WITH_PARCELPORT_MPI) OR HPXLocal_WITH_ASYNC_MPI)
+if((HPX_WITH_NETWORKING AND HPX_WITH_PARCELPORT_MPI)
+   AND NOT HPXLocal_WITH_ASYNC_MPI
+)
```

Comment:
Note that this is an oxymoron. This will never be true.

I think the correct thing would be to add `OMPI_IMPORTS` as a compile flag in HPXLocal, not as a config define. I'm guessing the `FIXME` is trying to say this as well... Does that sound reasonable?

Reply:
> Note that this is an oxymoron. This will never be true.

In this case we can remove it from this repository alltogether. 

> I think the correct thing would be to add `OMPI_IMPORTS` as a compile flag in HPXLocal, not as a config define. I'm guessing the `FIXME` is trying to say this as well... Does that sound reasonable?

What would that change? In the end it needs to be defined whenever mpi.h (from OpenMPI) is included.

Reply:
> > Note that this is an oxymoron. This will never be true.
> 
> In this case we can remove it from this repository alltogether.

Yes, it's enough to have it in HPXLocal.
 
> > I think the correct thing would be to add `OMPI_IMPORTS` as a compile flag in HPXLocal, not as a config define. I'm guessing the `FIXME` is trying to say this as well... Does that sound reasonable?
> 
> What would that change? In the end it needs to be defined whenever mpi.h (from OpenMPI) is included.

Multiple `-DOMPI_IMPORTS` on the command line will not conflict (as long as they don't try to set a value to it of course...). Alternatively, the define in the config should be conditional. Defining it with `-DOMPI_IMPORTS` has the benefit of not requiring `hpx/config.hpp` to be included before `mpi.h` though.

In any case, I think the define can be removed from here in all its possible forms.